app/controllers/books_controller.rb

```diff
@@ -1,14 +1,18 @@
 class BooksController < ApplicationController
-  before_action :set_book, only: %i[show edit update destroy]
+  before_action :set_book, only: %i[edit update destroy]
+  before_action :set_user
 
   def index
-    @books = Book.all
+    @user = set_user
```

Comment:
As we discussed, it's not always the same as current user - user can also see the books of any other user. But you are right: I do not need to set this method inside `index`, since I set it `before action`